src/rational/mat_q/arithmetic/mul_scalar.rs

```diff
@@ -0,0 +1,301 @@
+        let mat3 = MatQ::from_str("[[2,3],[3/2,6]]").unwrap();
+        let integer1 = Z::from(3);
+        let integer2 = Z::from(3);
+
+        let mat1 = mat1 * integer1;
+        let mat2 = integer2 * mat2;
+
+        assert_eq!(mat3, mat1);
+        assert_eq!(mat3, mat2);
+    }
+
+    /// Checks if scalar multiplication works fine for half owned/borrowed
+    #[test]
+    fn half_correctness() {
+        let mat1 = MatQ::from_str("[[2/3,1],[1/2,2]]").unwrap();
+        let mat2 = mat1.clone();
+        let mat3 = mat1.clone();
+        let mat4 = mat1.clone();
+        let mat5 = MatQ::from_str("[[2,3],[3/2,6]]").unwrap();
+        let integer1 = Z::from(3);
+        let integer2 = Z::from(3);
+
+        let mat1 = mat1 * &integer1;
+        let mat2 = &integer2 * mat2;
+        let mat3 = &mat3 * integer1;
+        let mat4 = integer2 * &mat4;
+
+        assert_eq!(mat5, mat1);
+        assert_eq!(mat5, mat2);
+        assert_eq!(mat5, mat3);
+        assert_eq!(mat5, mat4);
+    }
+
+    /// Checks if scalar multiplication works fine for different scalar types
+    #[test]
+    #[allow(clippy::erasing_op)]
+    fn different_types() {
+        let mat1 = MatQ::from_str("[[1/2],[0],[4]]").unwrap();
+        let mat2 = MatQ::from_str("[[2,5,6],[1,3,1]]").unwrap();
+        let mat3 = MatQ::from_str("[[1],[0],[8]]").unwrap();
+        let mat4 = MatQ::from_str("[[0],[0],[0]]").unwrap();
+        let mat5 = MatQ::from_str("[[-1/2],[0],[-4]]").unwrap();
+        let mat6 = MatQ::from_str("[[6,15,18],[3,9,3]]").unwrap();
+
+        assert_eq!(mat3, 2 * &mat1);
+        assert_eq!(mat4, 0 * &mat1);
+        assert_eq!(mat5, -1 * mat1);
+        assert_eq!(mat6, mat2 * 3);
+    }
+
+    /// Checks if scalar multiplication works fine for matrices of different dimensions
+    #[test]
+    fn different_dimensions_correctness() {
+        let mat1 = MatQ::from_str("[[1/2],[0],[4]]").unwrap();
+        let mat2 = MatQ::from_str("[[2,5/8,6],[1,3,1/7]]").unwrap();
+        let mat3 = MatQ::from_str("[[3/2],[0],[12]]").unwrap();
+        let mat4 = MatQ::from_str("[[6,15/8,18],[3,9,3/7]]").unwrap();
+        let integer = Z::from(3);
+
+        assert_eq!(mat3, &integer * mat1);
+        assert_eq!(mat4, integer * mat2);
+    }
+
+    /// Checks if matrix multiplication works fine for large values
+    #[test]
+    fn large_entries() {
+        let mat1 = MatQ::from_str(&format!("[[1],[{}],[1/{}]]", i64::MAX, i64::MAX)).unwrap();
+        let mat2 = MatQ::from_str("[[3]]").unwrap();
+        let mat3 = MatQ::from_str(&format!(
+            "[[3],[{}],[3/{}]]",
+            3 * i64::MAX as i128,
+            i64::MAX
+        ))
+        .unwrap();
+        let mat4 = MatQ::from_str(&format!("[[{}]]", 3 * i64::MAX as i128)).unwrap();
+        let integer1 = Z::from(3);
+        let integer2 = Z::from(i64::MAX);
+
+        assert_eq!(mat3, integer1 * mat1);
+        assert_eq!(mat4, integer2 * mat2);
+    }
+}
+
+#[cfg(test)]
+mod test_mul_q {
```

Comment:
Availability would here also be nice, if you add an implementation for `f32` and `f64`